### ISSUE TITLE
add status to the revisions command 

### ISSUE THREAD

**Issue by danfein** [cloudfoundry/cli#1958]
**Please provide details on the following items. Failure to do so may result in deletion of your feature request.** 

**What's the user value of this feature request?**
Additional information in the output of the cf revisions command to help folks reason about what happened at each revision, and make it easier to make choices when looking for a revision to rollback to.

**Who is the functionality for?**
Primarily human users, application developers / application operators.

**How often will this functionality be used by the user?**
Whenever an app rollback is required (hopefully never? ;)
It could be helpful in some troubleshooting workflows to see the history of changes to an app to correlate them to metrics or behavior changes.

**Who else is affected by the change?**
The re-ordering of the columns could break automation that is scraping the tables. 

**Is your feature request related to a problem? Please describe.**
The current revisions output is great and surfaces helpful information from the API in a more convenient way. The changes proposed would add more relevant information to support the ‘decide which revision’ use-case.

**Describe the solution you'd like**
Proposed: addition of status information columns
Proposed order columns 
• Revision number (newest first - mixed testing here, newest first helps scrapers, but human readers would have to scroll up to see newest if they have lots of revisions) - feedback welcome
• status (new) - (values such as deployed, canceled, not deployed, active)
• Description 
• Deployable (proposed in issue 1949)
• Revision GUID 
• Created on timestamp

The order was moved a bit given that Timestamp and guid run long, and given that you would know if you needed a guid or not based on the status/deployable columns, we prioritized running those columns first. (again, feedback welcome)

**Describe alternatives you've considered**
--

**Additional context**

```
$ cf revisions testing-rolling
Getting revisions for app testing-rolling in org meehp / space maahp as admin...
OK

revision  status            description                         deployable   revision guid:                        created on
4         deployed(active)  new droplet deployed                true         585bc3c1-3743-497d-88b0-403ad6b56d16  2020-03-01T01:33:58Z
3         canceled          rolled back to revision 1           true         7864380b-f798-480b-b145-6cc4e635b3f9  2020-02-01T01:33:58Z
2         deployed          new droplet deployed                false        5327427f-ce2e-4317-8626-f81fc9d7fd71  2020-01-01T01:33:58Z
1         not deployed      new environment variables deployed  true         585bc3c1-3743-497d-88b0-403ad6b56d16  2019-02-01T01:33:58Z
0         superseded        new droplet deployed                false        743cd6d1-d121-416c-90af-25949ceaa180  2017-02-01T01:33:58Z
```

**Comment by danfein** [comment]
@pivotalrobbie @heyjcollins FYI

**Comment by heyjcollins** [comment]
We've added specific details to our tracker story for this issue and pulled this into our backlog as a [**feature story**](https://www.pivotaltracker.com/story/show/173456346) that will be included in our next minor release.

Thanks @danfein !

**Comment by heyjcollins** [comment]
hey @danfein -
We ran into a snag with the "status" column/values while implementing these changes.

After a ton of discussion with CAPI folks (namely Chris Selzo), we've decided to drop the idea of a status column and replace it with an `active` (true or false) column. 

There are two reasons for this:
Our original idea to correlate a revision with a droplet, doesn't give a very useful status. 
The only thing it will tell you is if the droplet is staged or expired (and a few other less common states). 
It's not really the type of info that someone would want to use to decide whether to rollback to a revision, they would prefer to see which revision is currently active, which one is in the process of being deployed, etc.

So we looked into correlating revisions with a `deployment` instead of a droplet to get our status as the deployment endpoint actually does return the status values as defined in the original mockup you provided when you logged this issue.

However, there are numerous cases where a revision will not be associated with a deployment.
- app was originally pushed on an older version of cf-deployment with a CAPI that didn't support for `deployment` construct
- app was pushed, restarted, restaged without --strategy rolling (deployments are only created/associated with app pushed/updated with the rolling deployment strategy)

Given not all apps have been or will be pushed/updated using the rolling strategy, if we rely on the 'deployments' endpoint to provide values for the revisions>status column, all revisions (including the current active revision) may have a deployment status value of `not deployed` or perhaps if we altered that to something like `unknown` in the CLI client it might be slightly better --- but overall we believe this to be an extremely confusing UX.
Given the importance of clear understanding of which revision is currently live, which can be rolled back to, etc... we've decided against using the deployment endpoint

The status field would be confusing, unreliable, and/or not relevant if we use either approach. 

@selzoc brought the endpoint http://v3-apidocs.cloudfoundry.org/version/3.88.0/index.html#list-deployed-revisions-for-an-app to our attention. 
The output from that endpoint will allow us to figure out which revision is active. 
Knowing which revision is `active` (currently live) in combination with the value in the `deployable` column should provide the right amound of actionable information for making a rollback decision.

Side Note: The most recent revision cannot be guaranteed to be the active revision (hence the need for `active` column. This can be achieved via the following steps
1. update an existing app that's scaled to multiple instances with --strategy rolling
  - cf push appname --strategy rolling 
1. cancel the deployment before all the app instances have been rolled
  - cf cancel-deployment appname


So as such here's our alternative proposed `cf revisions` output:

**Comment by heyjcollins** [comment]
@danfein - due to some inconsistencies regarding the availability of of the status data we've decided to remove the `active` column and to decorate the revision number associated with the currently running app with `(deployed)`.
However, when an app is **not** running, `(deployed)` decorator will not be associated with any of the stopped app's revisions.

**Comment by reid47** [comment]
We recently released [v7.2.0](https://github.com/cloudfoundry/cli/releases/tag/v7.2.0), which further improves the `revisions` command. See Josh's comments for details. Thanks for your help!
